ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -60,19 +65,19 @@ export class WorkflowLogsViewer extends React.Component<WorkflowLogsViewerProps,
                         </p>
                     )}
                     {!this.state.error && this.podHasNoLogs() && <p>Pod did not output any logs.</p>}
-                    {this.state.lines.length > 0 && (
+                    {this.state.lineCount > 0 && this.logsObservable && (
                         <div className='log-box'>
                             <FullHeightLogsViewer
                                 source={{
                                     key: `${this.props.workflow.metadata.name}-${this.props.container}`,
-                                    loadLogs: () => Observable.from(this.state.lines),
+                                    loadLogs: identity(this.logsObservable),
```

Comment:
This change was introduced so that when strict types are added, we don't have to go back and fix this.  Inside the function (with strict typings enabled), the type of `logsObservable` would be `Observable<string> | undefined`.  By adding the conditional above and using the `identity` function, we can avoid that and resolve the type to `Observable<string>` like the API requires.